engine/tx_broadcast.go

```diff
@@ -0,0 +1,46 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chainstate"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+)
+
+func broadcastTransaction(ctx context.Context, tx *Transaction) error {
+	client := tx.Client()
+	chainstateSrv := client.Chainstate()
+
+	defer recoverAndLog(tx.Client().Logger())
```

Comment:
maybe this way 
```go
defer func() {
		if r := recover(); r != nil {
			tx.Client().Logger().Error().Str("txID", tx.ID).Msgf("panic in broadcastTransaction: %v", r)
			recoverAndLog(tx.Client().Logger())
		}
	}()
```

Reply:
The "not mine" function `recoverAndLog` is implemented as follows:
```golang
func recoverAndLog(log *zerolog.Logger) {
	if err := recover(); err != nil {
		log.Error().Msgf(
			"panic: %v - stack trace: %v", err,
			strings.ReplaceAll(string(debug.Stack()), "\n", ""),
		)
	}
}
```

I've just used it the same way it had been used previously.

Reply:
Maybe some new task to "Investigate potential risks of using `recoverAndLog` function ?

Reply:
We could use "named return params" and `"catch"` the panics inside some function but return an error 

```golang
func someFunc() (err error) {
    defer func() {
        if err := recover(); err != nil {
		//optional logging
                err = spverrors.SomeError.Wrap(err)
	}
    }
    //..normal flow
   
}
```

Reply:
I've added a JIRA task for that `SPV-1068`